### PR TITLE
Log when there are empty parse results

### PR DESCRIPTION
**Pull request by jez**
### Motivation

I want to be able to let users generate test cases for me.

This log line will let me see what file contents people are seeing when they get
empty parse results.

(Note that we already effectively put the entire file contents into the log
file, because we log every LSP protocol message, including the
`textDocument/didChange` requests, this just makes it easier to correlate
"failure condition detected" and "file contents".)

### Test plan

I tested this by running LSP with `--debug-log-file` locally and checking the
contents of the log file.

If we feel strongly, I can write a test for this, but it's a little annoying,
because it only shows up in LSP mode and in the log file. It'd be doable to try
to get the contents of that log file but require some originality.

### CODE REVIEW

**Comment by jez** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_L2PCppPOUgf0Wo
→ https://go/builds/bui_L2PCcuMN5ZYmSI

File: main/lsp/LSPTypechecker.cc [hunk 1]
```diff
@@ -247,7 +247,7 @@ vector<core::FileRef> LSPTypechecker::runFastPath(LSPFileUpdates &updates, Worke
     fast_sort(subset);
     subset.resize(std::distance(subset.begin(), std::unique(subset.begin(), subset.end())));
 
-    config->logger->debug("Taking fast path");
+    config->logger->debug("Running fast path");
```
Comment (aprocter) [inline main/lsp/LSPTypechecker.cc:250]: Was this change intentional? (I have no objection to it if so.)

File: hashing/hashing.cc [hunk 2]
```diff
@@ -27,8 +28,50 @@ pair<ast::ParsedFile, core::UsageHash> rewriteAST(const core::GlobalState &origi
...
+                writer.String(string(file.file.data(*lgs).source()));
+
+                writer.EndObject();
+            }
+
+            logger.debug(result.GetString());
+
+            core::GlobalStateHash invalid;
+            invalid.hierarchyHash = core::GlobalStateHash::HASH_STATE_INVALID;
+            return make_unique<core::FileHash>(move(*lgs->hash()), move(usageHash));
```
Comment (aprocter) [inline hashing/hashing.cc:71]: Did you mean to pass `invalid` in place of `*lgs->hash()` or something? (Otherwise I'm not sure what we're using `invalid` for.)

**Review by aprocter — APPROVED** [review]

File: main/lsp/LSPTypechecker.cc (hunk 1 above)
Comment (jez) [reply main/lsp/LSPTypechecker.cc:250]: yeah, there's another log line that's identical, and i wanted this to be different.

File: hashing/hashing.cc (hunk 2 above)
Comment (jez) [reply hashing/hashing.cc]: good catch, i fixed that, and i wrote a test that would have caught this mistake.
